ostd/src/sync/spin.rs

```diff
@@ -84,22 +107,20 @@ impl<T: ?Sized> SpinLock<T> {
     /// for compile-time checked lifetimes of the lock guard.
     ///
     /// [`lock`]: Self::lock
-    pub fn lock_arc(self: &Arc<Self>) -> ArcSpinLockGuard<T> {
-        let guard = disable_preempt();
+    pub fn lock_arc(self: &Arc<Self>) -> ArcSpinLockGuard<T, G> {
         self.acquire_lock();
         SpinLockGuard_ {
             lock: self.clone(),
-            inner_guard: InnerGuard::PreemptGuard(guard),
+            guard: G::guard(),
         }
     }
 
     /// Tries acquiring the spin lock immedidately without disabling the local IRQs.
```

Comment:
"without disabling local IRQs"? This should also be outdated documentation.

I think a lot of documentation in this file is outdated, we'd better go through it to find out all of them.